code-the-dream-react/src/App.js

```diff
@@ -1,33 +1,61 @@
-import * as React from 'react';
 import { useEffect } from 'react';
+import { useState } from 'react';
 import { TodoList } from './TodoList.js';
 import { AddTodoForm } from './AddTodoForm.js';
- 
-function useSemiPersistentState() {
-  const [todoList, setTodoList] = React.useState(() => {
-    const savedTodoList = localStorage.getItem('savedTodoList');
-    return savedTodoList ? JSON.parse(savedTodoList) : [];
-  });
+
+function App() {
+  const [todoList, setTodoList] = useState([]);
+  const [isLoading, setIsLoading] = useState(true);
+
   useEffect(() => {
-    localStorage.setItem('savedTodoList', JSON.stringify(todoList));
-  }, [todoList]);
+    // Fetch data only once when the component mounts
+    const fetchData = new Promise((resolve, reject) => {
+      // Simulate 2-second delay to resolve the Promise
+      setTimeout(() => {
+        resolve({ data: { todoList: [] } });
+      }, 2000);
+    });
 
-  return [todoList, setTodoList];
-}
+    // Use .then() to update the todoList state with the resolved data
+    fetchData
+      .then((result) => {
+        const updatedTodoList = result.data.todoList;
+        setTodoList(updatedTodoList);
+        setIsLoading(false); // Set isLoading to false after data is fetched
+      })
+      .catch((error) => {
+        console.error('Error:', error);
+        setIsLoading(false);
+      });
+  }, []); // Empty dependency array to fetch data only once
```

Comment:
for a better composition of your app component, I would advise you to pull this fetchData logic to its own function and just call the functions within this effect, this will improve the readability of your code.